api/spec/v1.yaml

```diff
@@ -2085,8 +2085,33 @@ components:
           description: An array containing details of the last 100 consensus blocks, indicating whether each block was signed by the validator. Only available when querying a single validator.
           items:
             allOf: [$ref: '#/components/schemas/ValidatorSignedBlock']
+        uptime:
+          allOf: [$ref: '#/components/schemas/ValidatorUptime']
+          description: The validator's uptime statistics for a period of time up to now.
       description: |
-        An validator registered at the consensus layer.
+        A validator registered at the consensus layer.
+
+    ValidatorUptime:
+      type: object
+      properties:
+        window_length:
+          type: integer
+          format: uint64
+          description: The length of the window this object describes, in blocks.
+        partial_length:
+          type: integer
+          format: uint64
+          description: The length of the partial windows within window_length, in blocks.
```

Comment:
```suggestion
        segment_length:
          type: integer
          format: uint64
          description: The length of the window segment, in blocks. We subdivide the window into segments of equal length and aggregate the uptime of each segment into `segment_uptimes`. Currently defaults to 1200 blocks, which is approximately 2 hours.
```

Reply:
I'm not very enthusiastic about `segment`, I considered using `chunk` or `subwindow` too. Feel free to change